e2e/favorites.spec.ts

```diff
@@ -1,13 +1,120 @@
+  await expectFavoritesPageLoaded(page)
+}
+
+async function expectProjectPageLoaded(
+  page: Page,
+  projectName: string,
+  projectId: string,
+) {
+  await page.waitForURL(entityUrlPathname(projectId))
+  await expect(page.getByRole('heading', { name: projectName })).toBeVisible()
+  await page.waitForURL(
+    `${entityUrlPathname(PUBLIC_PROJECT.id)}/discussion/default`,
+  )
+  await expect(page.getByRole('heading', { name: 'Discussion' })).toBeVisible()
+}
 
 test.describe('Favorites', () => {
   testAuth('should be visible', async ({ userPage }) => {
     await goToDashboard(userPage)
-    await userPage.getByLabel('Favorites', { exact: true }).click()
-    await expect(
-      userPage.getByRole('heading', { name: 'Your Favorites' }),
-    ).toBeVisible()
+    await goToFavorites(userPage)
   })
+
+  testAuth(
+    'should be able to favorite and unfavorite a project',
+    async ({ userPage }) => {
+      const addFavorite = userPage.getByLabel(
+        'Click the star to add this to your favorites',
+      )
+      const removeFavorite = userPage.getByLabel(
+        'Click the star to remove this from your favorites',
+      )
+      const projectLink = userPage.getByRole('link', {
+        name: PUBLIC_PROJECT.name,
+      })
+
+      await testAuth.step('user goes to public project', async () => {
+        await userPage.goto(entityUrlPathname(PUBLIC_PROJECT.id))
+        await waitForInitialPageLoad(userPage)
+        await expectProjectPageLoaded(
+          userPage,
+          PUBLIC_PROJECT.name,
+          PUBLIC_PROJECT.id,
+        )
+      })
+
+      await testAuth.step('user favorites project', async () => {
+        await expect(addFavorite).toBeVisible()
+        await expect(removeFavorite).not.toBeVisible()
+
+        await addFavorite.click()
+
+        await expect(addFavorite).not.toBeVisible()
+        await expect(removeFavorite).toBeVisible()
+      })
+
+      await testAuth.step(
+        'user views project on their favorites page',
+        async () => {
+          await goToFavorites(userPage)
+          await expect(projectLink).toBeVisible()
+        },
+      )
+
+      await testAuth.step(
+        'user navigates to project from favorites',
+        async () => {
+          await projectLink.click()
+
+          await expectProjectPageLoaded(
+            userPage,
+            PUBLIC_PROJECT.name,
+            PUBLIC_PROJECT.id,
+          )
+        },
+      )
+
+      await testAuth.step('user removes project as a favorite', async () => {
+        await expect(addFavorite).not.toBeVisible()
+        await expect(removeFavorite).toBeVisible()
+
+        await removeFavorite.click()
+
+        await expect(addFavorite).toBeVisible()
+        await expect(removeFavorite).not.toBeVisible()
+      })
+
+      await testAuth.step(
+        'User does not have project listed on their favorites page',
+        async () => {
+          await goToFavorites(userPage)
+          await userPage.reload()
```

Comment:
The page needs to be reloaded for the favorites list to update

Reply:
Can you add as a comment right above this line?

```suggestion
          // The page needs to be reloaded for the favorites list to update
          await userPage.reload()
```


At some point this might use a cache like react-query, where the reload won't be necessary--if/when that affects this integration test, the comment could help us find that and decide if it can be removed.